tests/client/test_client.py

```diff
@@ -263,6 +263,7 @@ def test_client_register(patch_post, compressed, monkeypatch, tmpdir):
         compressed=compressed,
         version_group_id=str(uuid.uuid4()),
         no_url=True,
+        idempotency_key="foo",
```

Comment:
There's no real place to add test coverage for this in core but we'll make sure it can be passed into `client_register`